caddyhttp/proxy/upstream_test.go

```diff
@@ -225,4 +227,38 @@ func TestParseBlock(t *testing.T) {
 			}
 		}
 	}
+
+	// tests for insecure skip verify
+	isv_tests := []struct {
+		config string
+		flag   bool
+	}{
+		// Test #1: without flag
+		{"proxy / localhost:8080 {\n health_check / \n}", false},
+
+		// Test #2: with flag
+		{"proxy / localhost:8080 {\n health_check / \n insecure_skip_verify \n}", true},
+	}
+
+	for i, test := range isv_tests {
+		upstreams, err := NewStaticUpstreams(caddyfile.NewDispenser("Testfile", strings.NewReader(test.config)))
+		if err != nil {
+			t.Error("Expected no error. Got:", err.Error())
+		}
+		for _, upstream := range upstreams {
+			staticUpstream, ok := upstream.(*staticUpstream)
+			if !ok {
+				t.Error("Type mismatch")
+			}
+			transport, ok := staticUpstream.HealthCheck.Client.Transport.(*http.Transport)
+			if !ok {
+				t.Error(reflect.TypeOf(transport.TLSClientConfig))
+				t.Error("Type mismatch")
+				continue
+			}
+			if test.flag != transport.TLSClientConfig.InsecureSkipVerify {
+				t.Error("Test case #" + string(i) + "failed")
```

Comment:
This will not have the intended output, I think. And a more useful error message is good. Instead, use `t.Errorf("test %d: expected transport.TLSClientConfig.InsecureSkipVerify=%v, got %v", i, ...)` (you get the idea)